Firestore/core/src/core/query.h

```diff
@@ -277,25 +277,38 @@ class Query {
   // sort at the end.
   std::vector<OrderBy> explicit_order_bys_;
 
-  // The memoized list of sort orders.
-  mutable std::vector<OrderBy> memoized_normalized_order_bys_;
-
   int32_t limit_ = Target::kNoLimit;
   LimitType limit_type_ = LimitType::None;
 
   absl::optional<Bound> start_at_;
   absl::optional<Bound> end_at_;
 
+  const std::shared_ptr<const Target> ToTarget(
```

Comment:
Can you try modifying this overload of `ToTarget()` to just return `Target` rather than `const std::shared_ptr<const Target>`? It looks like wrapping it in a `std::shared_ptr` is no longer needed now that you're using `ThreadSafeMemoizer`.